ethcore/src/client/test_client.rs

```diff
@@ -665,13 +673,18 @@ impl BlockChainClient for TestBlockChainClient {
 		self.receipts.read().get(&id).cloned()
 	}
 
-	fn logs(&self, filter: Filter) -> Vec<LocalizedLogEntry> {
+	fn logs(&self, filter: Filter) -> Result<Vec<LocalizedLogEntry>, BlockId> {
+		match self.error_on_logs.read().as_ref() {
+			Some(id) => return Err(id.clone()),
```

Comment:
Would it make sense to return both the block ID and the block hash?

Reply:
This `BlockId` contains `BlockId::Number` and `BlockId::Hash`, so it may be fine here -- the tester can choose which type of `BlockId` it wants to test. :)